models/chainlink/arbitrum/chainlink_arbitrum_ccip_fulfilled_transactions.sql

```diff
@@ -2,61 +2,36 @@
   config(
     
     alias='ccip_fulfilled_transactions',
-    partition_by=['date_month'],
+    partition_by=['date_start'],
     materialized='incremental',
     file_format='delta',
-    incremental_strategy='merge',
-    unique_key=['tx_hash', 'tx_index', 'node_address']
```

Comment:
why are we removing the unique keys and the corresponding test in the schema file?

if a model is materialized incrementally and set to merge, we need the unique keys to understand how the merge statement is written. without these key columns, wouldn't it just insert over and over? did you query the output CI tables to test the data?

Reply:
The reasoning for the deletion of unique keys and tests is looking into how we are doing ccip fulfilled tx's there was a lot of un needed data being joined and passed. Unlike the other fulfilled transaction spells for the other chainlink products this one is now taking the traces instead of logs which then causes issues for unique tests since there could/will be rows that have the same tx_hash, node_address etc since there could be multiple of the same trace in the same tx. This was found after testing and bringing back incorrect data in how it was set before. I have done some testing on this PR and didnt find it inserting over and over. I suppose a question to you would be in this case is this the correct way to label it?

Reply:
i think you have a few options:
1. add more keys to unique key list (can you add `trace_address` to help differentiate within tx?)
2. change to append-only strategy, but then we need to consider how to avoid writing data multiple times (for instance, incremental runs reading back 7 days would load those 7 days over and over)
3. change to table materialization, where a full refresh happens each time, rather than looking to keep building incrementally

Reply:
just looked into the first option and I think this may be the best way. Ill add this in and add the tests back for this as well. Thanks for the recommendations.